### PR TITLE
Fix/forget cases on logout cu 1vzderh

### PR DESCRIPTION
**Pull request by jlarsson**
## Explain the changes you’ve made

CaseContext is more aware of user and user signing out

## Explain why these changes are made

Attempt to prevent bug related to user could see cases belonging to previously signed in user.
https://sharing.clickup.com/t/h/1vzderh/TRF9NPRQ0PHCDPR

## Explain your solution

- Breakout of dense logic in CaseContext.tsx to CaseContextHook.ts'
- Contexts (CaseState, CaseDispatch) have Null implementations when user is not signed in
- Reducer is reset based on sign out status

## How to test

Concrete example:

1. Checkout this branch
2. Swap the application to run storybook
3. Fire upp the simulator by running the command `yarn ios`

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1vzderh Bugg: case data raderas inte när en användare är utloggad](https://app.clickup.com/t/1vzderh)

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/store/CaseContext.tsx [hunk 1]
```diff
@@ -46,154 +45,38 @@ interface CaseProviderProps {
...
-      });
-    }
-  }, [pollLoop, state.isPolling, user]);
+  const lockDown = !isSignedIn;
+  const providedState = lockDown
+    ? createNullContextState()
+    : createContextState();
+  const providedDispatch = lockDown ? createNullDispatch() : createDispatch();
 
   useEffect(() => {
```
[the remark this answers is not in the document]
Comment (jlarsson) [reply source/store/CaseContext.tsx:67]: fixed - also had to wrap reset() in useCallback()

File: source/store/CaseContext.tsx [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
-  createCase as create,
-  deleteCase as remove,
-  fetchCases as fetch,
-  pollCase,
-} from "./actions/CaseActions";
 
-import { replaceCaseItemText } from "../containers/Form/hooks/textReplacement";
+import USER_AUTH_STATE from "app/types/UserAuthTypes";
+import useCaseState from "./CaseContext/CaseContextHooks";
+import { CurrentRenderContext } from "@react-navigation/native";
```
[the remark this answers is not in the document]
Comment (jlarsson) [reply source/store/CaseContext.tsx]: Fixed (this and a few other lint warnings)

**Comment by karlknowit** [comment]
Closing in favour for other solution:
https://github.com/helsingborg-stad/app-mitt-helsingborg/pull/646

**Pull request closed without merging**